### PR TITLE
New and hopefully improved with fewer steps

### PR DESCRIPTION
**Pull request by jobergum**
@bjorncs  For you

### CODE REVIEW

**Comment by jobergum** [comment]
Please review wording/issues @kkraune and @bjorncs  will complete the guide and make sure it works in a fresh aws setup.

File: documentation/vespa-quick-start-multinode-aws.html [hunk 1]
```diff
@@ -23,10 +23,10 @@ <h2 id="launch-instances">Launch instances on <a href="aws.amazon.com/ec2">AWS E
 	 which will limit access to the installation and allow our launced EC2 instances to talk to each other without restrictions.
```
Comment (kkraune) [inline documentation/vespa-quick-start-multinode-aws.html]: broken link two lines above - use https://aws.amazon.com/ec2/

Comment (kkraune) [inline documentation/vespa-quick-start-multinode-aws.html]: _First we'll need to_ is implicit, remove

File: documentation/vespa-quick-start-multinode-aws.html [hunk 2]
```diff
@@ -23,10 +23,10 @@ <h2 id="launch-instances">Launch instances on <a href="aws.amazon.com/ec2">AWS E
 	 which will limit access to the installation and allow our launced EC2 instances to talk to each other without restrictions. 
          See <a href="securing-your-vespa-installation.html">more about securing Vespa installations</a>.</p> 
 	 <ul>
-	<li><p>Allow access to inbound TCP port 8080 (feed & search interface) from your source IP range</p> </li>
-	<li><p>Allow access tp inbound for all TCP traffic on port range 0 - 65535 for members of the security group. 
-	This is done by first creating the security group with basic access from your source IP range to port 8080 for search and feed then once saved edit it with an 
-	additional inbound rule which uses the security group name as source</p>
+	<li><p>Allow access to inbound TCP port <samp>8080</samp> (feed &amp; search port) from your source IP range.</p> </li>
+	<li><p>Allow access tp inbound for all TCP traffic on port range <samp>0 - 65535</samp> for members of the security group.
```
Comment (kkraune) [inline documentation/vespa-quick-start-multinode-aws.html]: tp -> to

File: documentation/vespa-quick-start-multinode-aws.html [hunk 3]
```diff
@@ -43,91 +43,47 @@ <h2 id="launch-instances">Launch instances on <a href="aws.amazon.com/ec2">AWS E
 <h2 id="install-vespa">Install and run Vespa</h2>
```
Comment (kkraune) [inline documentation/vespa-quick-start-multinode-aws.html]: see _the_ sizing guide

File: documentation/vespa-quick-start-multinode-aws.html [hunk 4]
```diff
@@ -43,91 +43,47 @@ <h2 id="launch-instances">Launch instances on <a href="aws.amazon.com/ec2">AWS E
...
-	ssh -i my-aws.key centos@$i mv tmp.env.txt $VESPA_HOME/conf/vespa/default-env.txt;\
-  done
-</pre>
-<p>
-Vespa requires that the <em>'hostname'</em> command returns the exact same name as registered in DNS, e.g by <em>nslookup $(hostname) |grep Name</em>.
-If it does not match you can update it on CentOs 7 by running : </p>
+$ ssh-agent add my-aws-key.pem
+$ ssh -A centos@ec2-18-1-1-1.us-east-2.compute.amazonaws.com
+</pre> 
+<p>Now we are inside the AWS EC2 cloud and we can start boostraping our system install by downloading the saws bootstrap script which will install vespa and do basic system configuration for you:</p>
```
Comment (kkraune) [inline documentation/vespa-quick-start-multinode-aws.html]: saws -> aws

**Comment by kkraune** [comment]
All right! LGTM after addressing a few comments

**Comment by kkraune** [comment]
(after @bjorncs merging I will create a PR for less verbiage kraune-style, the content of this dos is great!)

File: documentation/vespa-quick-start-multinode-aws.html [hunk 5]
```diff
@@ -23,10 +23,10 @@ <h2 id="launch-instances">Launch instances on <a href="aws.amazon.com/ec2">AWS E
 	 which will limit access to the installation and allow our launced EC2 instances to talk to each other without restrictions. 
          See <a href="securing-your-vespa-installation.html">more about securing Vespa installations</a>.</p> 
 	 <ul>
-	<li><p>Allow access to inbound TCP port 8080 (feed & search interface) from your source IP range</p> </li>
-	<li><p>Allow access tp inbound for all TCP traffic on port range 0 - 65535 for members of the security group. 
-	This is done by first creating the security group with basic access from your source IP range to port 8080 for search and feed then once saved edit it with an 
-	additional inbound rule which uses the security group name as source</p>
+	<li><p>Allow access to inbound TCP port <samp>8080</samp> (feed &amp; search port) from your source IP range.</p> </li>
+	<li><p>Allow access tp inbound for all TCP traffic on port range <samp>0 - 65535</samp> for members of the security group. 
+	This is done by first creating the security group with basic access from your source IP range to port <samp>8080</samp> for search and feed then once saved edit it with an
```
Comment (bjorncs) [inline documentation/vespa-quick-start-multinode-aws.html]: Split into multiple sentences.

File: documentation/vespa-quick-start-multinode-aws.html [hunk 6]
```diff
@@ -23,10 +23,10 @@ <h2 id="launch-instances">Launch instances on <a href="aws.amazon.com/ec2">AWS E
...
          See <a href="securing-your-vespa-installation.html">more about securing Vespa installations</a>.</p> 
 	 <ul>
-	<li><p>Allow access to inbound TCP port 8080 (feed & search interface) from your source IP range</p> </li>
-	<li><p>Allow access tp inbound for all TCP traffic on port range 0 - 65535 for members of the security group. 
-	This is done by first creating the security group with basic access from your source IP range to port 8080 for search and feed then once saved edit it with an 
-	additional inbound rule which uses the security group name as source</p>
+	<li><p>Allow access to inbound TCP port <samp>8080</samp> (feed &amp; search port) from your source IP range.</p> </li>
+	<li><p>Allow access tp inbound for all TCP traffic on port range <samp>0 - 65535</samp> for members of the security group. 
+	This is done by first creating the security group with basic access from your source IP range to port <samp>8080</samp> for search and feed then once saved edit it with an 
+	additional inbound rule which uses the security group name as source.</p>
```
Comment (bjorncs) [inline documentation/vespa-quick-start-multinode-aws.html]: group name -> group id

File: documentation/vespa-quick-start-multinode-aws.html [hunk 7]
```diff
@@ -43,91 +43,47 @@ <h2 id="launch-instances">Launch instances on <a href="aws.amazon.com/ec2">AWS E
...
-  done
-
-[centos@ip-172-1-1-1 ~]$ for i in $(cat vespa-instances.txt); do \
-	ssh -i my-aws.key centos@$i mv tmp.env.txt $VESPA_HOME/conf/vespa/default-env.txt;\
-  done
-</pre>
-<p>
-Vespa requires that the <em>'hostname'</em> command returns the exact same name as registered in DNS, e.g by <em>nslookup $(hostname) |grep Name</em>.
-If it does not match you can update it on CentOs 7 by running : </p>
+$ ssh-agent add my-aws-key.pem
```
Comment (bjorncs) [inline documentation/vespa-quick-start-multinode-aws.html]: `ssh-add my-aws-key.pem`

File: documentation/vespa-quick-start-multinode-aws.html [hunk 8]
```diff
@@ -43,91 +43,47 @@ <h2 id="launch-instances">Launch instances on <a href="aws.amazon.com/ec2">AWS E
...
+$ ssh -A centos@ec2-18-1-1-1.us-east-2.compute.amazonaws.com
+</pre> 
+<p>Now we are inside the AWS EC2 cloud and we can start boostraping our system install by downloading the saws bootstrap script which will install vespa and do basic system configuration for you:</p>
 <pre>
-[centos@ip-172-1-1-1 ~]$ for i in $(cat vespa-instances.txt); do \
-	ssh -i my-aws.key centos@$i sudo hostnamectl set-hostname $i;\
-  done
+$ curl -s https://raw.githubusercontent.com/vespa-engine/sample-apps/bergum/aws-bootstrap/aws_bootstrap.sh  > aws_bootstrap.sh
```
Comment (bjorncs) [inline documentation/vespa-quick-start-multinode-aws.html]: URL seems incorrect (contains bergum).

Comment (jobergum) [reply documentation/vespa-quick-start-multinode-aws.html]: yes, it's on a branch yet.

File: documentation/vespa-quick-start-multinode-aws.html [hunk 9]
```diff
@@ -43,91 +43,47 @@ <h2 id="launch-instances">Launch instances on <a href="aws.amazon.com/ec2">AWS E
...
 
 <p> Create a hostfile listing the hosts in the system
-(<em>vespa-instances.txt</em>) to simplify the process multinode
-install process.
+(<samp>hosts.txt</samp>) to simplify the multinode install process.
 </p>
 <pre class="brush: cli">
-$ cat vespa-instances.txt
+$ cat hosts.txt
 ip-172-1-1-1.us-east-2.compute.internal
```
Comment (bjorncs) [inline documentation/vespa-quick-start-multinode-aws.html]: It is not obvious for me what these names are and where to find them. I think we should mention that these are the private dns addresses for the EC2 instances.

File: documentation/vespa-quick-start-multinode-aws.html [hunk 10]
```diff
@@ -43,91 +43,47 @@ <h2 id="launch-instances">Launch instances on <a href="aws.amazon.com/ec2">AWS E
 <h2 id="install-vespa">Install and run Vespa</h2>
 
 <p> Create a hostfile listing the hosts in the system
```
Comment (bjorncs) [inline documentation/vespa-quick-start-multinode-aws.html]: Move the section of creating a host file to after ssh'ing into the Vespa admininstration EC2 instance.

File: documentation/vespa-quick-start-multinode-aws.html (hunk 2 above)
Comment (bjorncs) [inline documentation/vespa-quick-start-multinode-aws.html]: Specify that the security group must allow inbound ssh connections.

File: documentation/vespa-quick-start-multinode-aws.html [hunk 11]
```diff
@@ -43,91 +43,47 @@ <h2 id="launch-instances">Launch instances on <a href="aws.amazon.com/ec2">AWS E
...
-
-<li><p><strong>Start the Vespa config server:</strong>
-Start the Vespa config server on the designated admin host.
-It needs to be up and running
-before deploying your application (wait five seconds after start, then proceed to next step):</p>
+<p>Now we are ready to install and start vespa, the script expects a single hostname as argument which should be the name of the administration/configuration server, in 
+our case we chose to use <samp>ip-172-1-1-1.us-east-2.compute.internal</samp></p>
 <pre>
-[centos@ip-172-1-1-1 ~]$ sudo service vespa-configserver start
+$ for host in $(cat hosts.txt); do ssh -f $host "sudo bash aws_bootstrap.sh ip-172-1-1-1.us-east-2.compute.internal"; done
```
Comment (bjorncs) [inline documentation/vespa-quick-start-multinode-aws.html]: Change to `for host in $(cat hosts.txt); do ssh -f $host "sudo bash aws_bootstrap.sh $(localhost)"; done`

File: documentation/vespa-quick-start-multinode-aws.html [hunk 12]
```diff
@@ -43,91 +43,47 @@ <h2 id="launch-instances">Launch instances on <a href="aws.amazon.com/ec2">AWS E
...
-<li><p><strong>Configure the multi-node application:</strong> An example application is found at
+<p>Now we have bootstraped our instances and it's time to configure Vespa using an example <a href="cloudconfig/application-packages.html">application package</a> as starting point.</p>
+<p><strong>Configure the multi-node application:</strong> An example application is found at
 <a href="https://github.com/vespa-engine/sample-apps/tree/master/basic-search">github.com/vespa-engine/sample-apps/basic-search</a>.
-Install git and clone a copy:</p>
 <pre>
-[centos@ip-172-1-1-1 ~]$ sudo yum -y install git 
-[centos@ip-172-1-1-1 ~]$ git clone https://github.com/vespa-engine/sample-apps.git 
-[centos@ip-172-1-1-1 ~]$ cd sample-apps/basic-search
+$ export VESPA_HOME=/opt/vespa; export PATH=$PATH:$VESPA_HOME/bin
```
Comment (bjorncs) [inline documentation/vespa-quick-start-multinode-aws.html]: Somewhere state that the following steps should be performed on the admin node.

File: documentation/vespa-quick-start-multinode-aws.html [hunk 13]
```diff
@@ -184,45 +143,54 @@ <h2 id="configure-vespa">Configure and deploy your Vespa application</h2>
...
 </pre>
-</li>
+<p>The <a href="reference/services.html"><samp>services.xml</samp></a> maps services to node aliases. 
+We have one <samp>configserver</samp>, two stateless jdisc search &amp; feed and two content nodes storing our data with a configured redundancy of 2.</p>
 
-<li><p><strong>Deploy the application :</strong></p>
+<p><strong>Deploy the application :</strong></p>
 <pre>
-[centos@ip-172-1-1-1 ~]$ vespa-deploy prepare src/main/application && vespa-deploy activate 
+vespa-deploy prepare src/main/application &amp;&amp; vespa-deploy activate
```
Comment (bjorncs) [inline documentation/vespa-quick-start-multinode-aws.html]: The config server was not started by the aws_bootstrap.sh script :/ I had to start it manually by `sudo service vespa-configserver start`.

**Comment by bjorncs** [comment]
Consider making the hostname placeholders in the shell commands and xml easier to identify by e.g. making them bold.

File: documentation/vespa-quick-start-multinode-aws.html [hunk 14]
```diff
@@ -184,45 +143,54 @@ <h2 id="configure-vespa">Configure and deploy your Vespa application</h2>
...
+java -jar $VESPA_HOME/lib/jars/vespa-http-client-jar-with-dependencies.jar \
   --file music-data-feed.json --host localhost --port 8080
 </pre>
-</li>
 
-<li><p><strong>Run a query against the containers</strong></p>
+<p><strong>Run a query against the search containers</strong></p>
+<p>Replace hostnames with your hostnames for which you have configured <samp>jdisc</samp> to run</p>
 <pre>
 $ curl -s http://ip-172-1-1-1.us-east-2.compute.internal:8080/search/?query=bad | python -m json.tool
```
Comment (bjorncs) [inline documentation/vespa-quick-start-multinode-aws.html]: Consider replacing the hostname with the alias defined in above hosts.xml file

**Comment by kkraune** [comment]
I added a commit to address all feedback, will merge now so we can iterate

**Pull request merged**